server/src/test/java/org/elasticsearch/action/admin/cluster/allocation/DesiredBalanceResponseTests.java

```diff
@@ -50,6 +55,29 @@ private DesiredBalanceStats randomStats() {
         );
     }
 
+    private ClusterBalanceStats randomClusterBalanceStats() {
+        return new ClusterBalanceStats(
+            randomBoolean()
+                ? Map.of(DiscoveryNodeRole.DATA_CONTENT_NODE_ROLE.roleName(), randomTierBalanceStats())
+                : randomSubsetOf(
+                    List.of(
+                        DiscoveryNodeRole.DATA_HOT_NODE_ROLE,
+                        DiscoveryNodeRole.DATA_WARM_NODE_ROLE,
+                        DiscoveryNodeRole.DATA_COLD_NODE_ROLE,
+                        DiscoveryNodeRole.DATA_FROZEN_NODE_ROLE
+                    )
+                ).stream().map(DiscoveryNodeRole::roleName).collect(toMap(identity(), ignore -> randomTierBalanceStats()))
+        );
+    }
+
+    private ClusterBalanceStats.TierBalanceStats randomTierBalanceStats() {
+        return new ClusterBalanceStats.TierBalanceStats(
+            new ClusterBalanceStats.MetricStats(randomDouble(), randomDouble(), randomDouble(), randomDouble(), randomDouble()),
```

Comment:
Could we create these from a random collection of nodes? I'm particularly interested in having test coverage for the case where one of the tiers is empty, which I think will give us some `NaN` values.